packages/components/src/progressive-disclosure-panel/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import noop from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ProgressiveDisclosurePanelTitle from './title';
+
+const ProgressiveDisclosurePanel = ( props ) => {
```

Comment:
I agree with you that this is confusing. It was [raised previously](https://github.com/WordPress/gutenberg/pull/32392#issuecomment-854628226), and [again](https://github.com/WordPress/gutenberg/pull/32392#issuecomment-856630475) when the behaviour was reverted to an earlier state when trying to nail down how the panel should behave.

I'd still like to see us come up with a better user experience here. An earlier suggestion was to disable a default control's menu item if it had been reset. It would be re-enabled once the default control's value was adjusted/set again. Another option could be to have multiple states for each menu item. Something like:

> There would be three menu item states:
>
> 1. No or hidden icon. Control is not shown and has no value.
> 2. Eye or check icon. Control is shown but has no value.
> 3. Check or reset/clear icon. Control is shown and has a value.

Open to suggestions. 

@jasmussen Do you have any thoughts now some time has passed since we last looked at this?

Reply:
I still think we should look for the simplest possible solution for the first version of this and then get a better feel in practice for how much of a problem this is.

Could we just remove the padding control if it's empty and you uncheck it? De-select and reselect the block and it's back, still empty. That's sort of what [the prototype](https://g2-components.xyz/?path=/story/designtools-presentation-typographypanel--default) does, even if it isn't explicitly explored.

Since I've sent you on so many wild chases on this one, I'd appreciate @mtias thoughts, as I know he worked on the initial prototype also.

Reply:
We did explore [having default controls be removed when deselected](https://github.com/WordPress/gutenberg/pull/32392#issuecomment-855513500). Essentially, they were only displayed by default on the initial selection or re-selection of the block. This made their behaviour more consistent with other controls and did simplify the code as well.

Whatever we decide here will be easy enough to work into the other updates and great suggestions Jorge has given.